pkg/services/queryhistory/database.go

```diff
@@ -41,6 +43,79 @@ func (s QueryHistoryService) createQuery(ctx context.Context, user *models.Signe
 	return dto, nil
 }
 
+func (s QueryHistoryService) searchQueries(ctx context.Context, user *models.SignedInUser, query SearchInQueryHistoryQuery) ([]QueryHistoryDTO, error) {
+	var queries []QueryHistoryDTO
+
+	if len(query.DatasourceUIDs) == 0 {
+		return nil, errors.New("no selected data source for query history search")
+	}
+
+	if query.Page <= 0 {
+		query.Page = 1
+	}
+
+	if query.Limit <= 0 {
+		query.Limit = 100
+	}
+
+	if query.Sort == "" {
+		query.Sort = "time-desc"
+	}
+
+	err := s.SQLStore.WithDbSession(ctx, func(session *sqlstore.DBSession) error {
+		sql := `SELECT
+			query_history.uid,
+			query_history.datasource_uid,
+			query_history.created_by,
+			query_history.created_at as "created_at",
+			query_history.comment,
+			query_history.queries,
+		`
+
+		if query.OnlyStarred {
+			sql = sql + `1 as "starred"
+				FROM query_history
+				INNER JOIN query_history_star ON query_history_star.query_uid = query_history.uid
+			`
+		} else {
+			sql = sql + `IIF(query_history_star.query_uid IS NULL, false, true) AS starred
+				FROM query_history
+				LEFT JOIN query_history_star ON query_history_star.query_uid = query_history.uid
+			`
+		}
+
+		sql = sql + `WHERE query_history.org_id = ? AND query_history.created_by = ? AND query_history.queries LIKE ? AND query_history.datasource_uid IN (?` + strings.Repeat(",?", len(query.DatasourceUIDs)-1) + `)
```

Comment:
Need to use the [LikeStr](https://github.com/grafana/grafana/blob/e814e7364bebe80c61c2439e9f54f908a7dcce87/pkg/services/sqlstore/migrator/dialect.go#L25) from the dialect because there's differences between databases. See for example  https://github.com/grafana/grafana/blob/e814e7364bebe80c61c2439e9f54f908a7dcce87/pkg/services/libraryelements/writers.go#L61

Reply:
Updated in https://github.com/grafana/grafana/pull/45932/commits/9996cf8ff6f0978cf11c4632b54546511cec0896 with some refactor to previous commit.